utils/include/tlv.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdint.h>
+#include <stdbool.h>
+
+/**
+ * @brief Write data to storage.
+ * 
+ * @param ctx Storage context.
+ * @param offset Location where to place the date.
+ * @param data Data to save.
+ * @param data_size Number of bytes to save.
+ * 
+ */
+typedef int (*tlv_storage_write)(void *ctx, uint32_t offset, uint8_t *data, uint32_t data_size);
+
+/**
+ * @brief Read data from storage.
+ * 
+ * @param ctx Storage context.
+ * @param offset Location where to find the date.
+ * @param data Buffor where data will be placed.
+ * @param data_size Number of bytes to read.
+ * 
+ */
+typedef int (*tlv_storage_read)(void *ctx, uint32_t offset, uint8_t *data, uint32_t data_cap);
+
+typedef struct tlv_ctx {
+	struct storage_impl {
```

Comment:
minor: the name `module_impl` is usaly used when there is a `module_ifc`
I think in this module the name like `sorage_s` or `tlv_sotrage` would be better

Reply:
changed to tlv_storage